drivers/soundwire/stream.c

```diff
@@ -1655,12 +1659,27 @@ static int _sdw_deprepare_stream(struct sdw_stream_runtime *stream)
 			return ret;
 		}
 
+		multi_lane_bandwidth = 0;
+
+		list_for_each_entry(p_rt, &m_rt->port_list, port_node) {
+			if (!p_rt->lane)
+				continue;
+
+			bandwidth = m_rt->stream->params.rate * hweight32(p_rt->ch_mask) *
+				m_rt->stream->params.bps;
+			multi_lane_bandwidth += bandwidth;
+			bus->lane_used_bandwidth[p_rt->lane] -= bandwidth;
+		}
 		/* TODO: Update this during Device-Device support */
-		bus->params.bandwidth -= m_rt->stream->params.rate *
-			m_rt->ch_count * m_rt->stream->params.bps;
+		bandwidth = m_rt->stream->params.rate * m_rt->ch_count * m_rt->stream->params.bps;
+		bus->params.bandwidth -= bandwidth - multi_lane_bandwidth;
+
+		/* No need to compute params since bus->params.bandwidth is not changed */
+		if (multi_lane_bandwidth == bandwidth)
+			compute_params = false;
 
 		/* Compute params */
-		if (bus->compute_params) {
+		if (compute_params && bus->compute_params) {
```

Comment:
The one question I have is whether all this works with the 192kHz 8ch case, where we will have to put 4 channels on one link and 4 channels on the other.

Reply:
I think it should work, but I don't have an environment to test. I have a question, why don't we use 1 link with 4 lanes?

Reply:
sorry @bardliao, I meant "The one question I have is whether all this works with the 192kHz 8ch case, where we will have to put 4 channels on one lane and 4 channels on a second lane."

Reply:
> sorry @bardliao, I meant "The one question I have is whether all this works with the 192kHz 8ch case, where we will have to put 4 channels on one lane and 4 channels on a second lane."

Yes, I believe it will work.